Fix #8093: Build+Refit changed game-state in command test run, and thus caused desyncs

frosch123 opened the pull request:
Still did not test this properly :)

glx22 reviewed (commented):
All my tests passed.

LordAro commented:
Needs a rebase now the ogfx change has been split out

frosch123 committed:
Codechange: Unify the tests whether build+refit is in simulation-test or real-run.

frosch123 committed:
Fix: When build+refit an engine, do not refit any free wagons that may get attached.

frosch123 committed:
Fix #8093: Build+Refit changed game-state in command test run, and thus caused desyncs.
Use DC_AUTOREPLACE for actions that shall be reversibe, in this case:
- Do not rearrange free wagons in test-run.
- Do not discard OrderBackups.
The latter was not triggered by actual auto-replace, since it does not set a 'user'.
